models/user.js

```diff
@@ -19,13 +19,13 @@ const UserSchema = new Schema({
         type: String,
         required: true
     },
-    role: {
-        type: [role.schema],
-        required: true
-    },
+    role: [{
+        type: role.schema.Types.ObjectId,
+        ref: 'Name'
```

Comment:
I don't think we have a model named `Name`
`{type: mongoose.Schema.Types.ObjectId, ref: 'Role'},`